src/+matlode/Integrator.m

```diff
@@ -2,51 +2,100 @@
     %INTEGRATOR Summary of this class goes here
     %   Detailed explanation goes here
     
+    properties (Abstract, Constant)
+        PartitionMethod
+    end
+    
     properties (Abstract, SetAccess = immutable)
-        c
-        adaptable
+        Adaptive
+        Datatype
     end
     
-    methods (Abstract)
-        
-        %integrate 
-        sol = integrate(obj, func, tspan, y0, varargin)
-        opts = matlodeSets(obj, varargin)
-        
+    methods (Abstract, Access = protected)
+        [t, y, stats] = timeLoop(obj, f, tspan, y0, opts, varargin);
+    end
+    
+    methods
+        function sol = integrate(obj, f, tspan, y0, varargin)
+            
+            if(length(tspan) < 2)
+                error('Unsupported tspan entry');
+            end
+            
+            if ~issorted(tspan) && ~issorted(tspan, 'descend')
+                error('tspan must be sorted in order')
+            end
+            
+            %TODO
+            %optimize options for when only options struct is given
+            
+            %Create options
+            p = inputParser;
+            opts = obj.matlodeSets(p, varargin{:});
+            
+            if isempty(opts.Dense) && length(tspan) > 2 && isa(opts.StepSizeController, 'matlode.stepsizecontroller.Fixed')
+                error('IntegrateTo is not supported for fixed step size, please use DenseOutput instead')
+            end
+            
+            t = [];
+            y = [];
+            stats = [];
+            
+            %Compute
+            if isa(f, 'function_handle') && ~obj.PartitionMethod
+                
+                [t, y, stats] = obj.timeLoop(f, tspan, y0, opts);
+                
+            elseif isa(f, 'cell') && cellfun(@(x) isa(x, 'function_handle'), f)
+                
+                %determine size of partition
+                if length(f) == 1 && ~obj.PartitionMethod
+                    [t, y, stats] = obj.timeLoop(f{1}, tspan, y0, opts);
+                elseif length(f) > 1
+                    %TODO
+                    %paritioning setup
+                    
+                else
+                    error('Need at least one function provided')
```

Comment:
I think function here is ambigous. maybe say

error('Partitioned system solves, require atleast one partition to be provided');